templates/funcs.go

```diff
@@ -172,11 +172,20 @@ func newArrayData(sliceType string, getter string, key string) ArrayData {
 }
 
 const arrayTpl = `
-{{ .SliceName }} := make(utils.{{ .SliceType }}, len({{ .Getter }}))
-for {{ .IndexName }}, {{ .ItemName }} := range {{ .Getter }} {
-	{{ .SliceName }}[{{ .IndexName }}] = {{ .ItemName }}
+
+{{ .Key }}Length := len({{ .Getter }})
+o.AddInt("{{ .Key }}_length", {{ .Key }}Length)
+
+if {{ .Key }}Length > 100 {
+	{{ .Key }}Length = 100
+}
```

Comment:
in this case, let's also rename the array log item key `{{ .Key }}` to `first_100_{{ .Key }}`